src/auth/auth.controller.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  CACHE_MANAGER,
+  Controller,
+  Get,
+  HttpCode,
+  HttpStatus,
+  Inject,
+  Post,
+  Redirect,
+  Req,
+  Res,
+  UseGuards,
+} from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { CurrentUser } from './decorator/current-user.decorator';
+import { CurrentUserDto } from './dto/current-user.dto';
+import { AuthService } from './auth.service';
+import { JwtPayload } from '../common/interfaces/jwt-payload.interface';
+import { Cache } from 'cache-manager';
+
+@Controller('auth')
+export class AuthController {
+  constructor(
+    private readonly authService: AuthService,
+    @Inject(CACHE_MANAGER) private cacheManager: Cache,
+  ) {}
+
+  /**
+   * Google Login - Redirect path /users/google/redirect
+   */
+  @UseGuards(AuthGuard('google'))
+  @HttpCode(HttpStatus.OK)
+  @Get('google')
+  signInWithGoogle(): void {
+    return;
+  }
+
+  /**
+   * Google Login - 로그인한 유저 정보 저장
+   * @param currentUser { accessToken, name, email, baseUrl }
+   * @returns cookie (accessToken, refreshToken) redirect 초기 화면 url
+   */
+  @UseGuards(AuthGuard('google'))
+  @HttpCode(HttpStatus.OK)
+  @Get('google/redirect')
+  @Redirect('http://localhost:8080', 301)
+  async signInWithGoogleRedirect(
+    @CurrentUser() { ...user }: CurrentUserDto,
```

Comment:
```suggestion
    @CurrentUser() user: CurrentUserDto,
```

구조분해할당 내에 전개 구문으로 모든 키밸류를 꺼낸다면 그냥 매개변수로 사용해도 좋을 거 같아요!